draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -527,12 +556,85 @@ To ensure readability and replicability, the examples in this specification do
 not contain decoy digests unless explicitly stated. For an example
 with decoy digests, see (#example-simple_structured).
 
+### Recursive Disclosures {#recursive_disclosures}
+
+The algorithms above are compatible with "recursive disclosures", in which one
+selectively disclosed field reveals the existence of more selectively
+disclosable fields.  For example, consider the following JSON structure:
+
+```json
+{
+    "family_name", "Möbius",
+    "nationalities": ["DE", "FR", "UK"] 
+}
+```
+
+The issuer may wish to allow the holder to reveal no nationalities at all, but
+require that if any nationalities are disclosed, then "DE" must be disclosed.
+This can be accomplished by first making the "FR" and "UK" entries within
+the array selectively disclosable, and then the whole "nationalities" field.
+
+``` ascii-art
+{
+    "family_name", "Möbius",
+    "nationalities": [
+        "DE", 
+        { "...": "Yr1KUNYXI6WzJKhP4Hwyu9Q1ejubVCcJRG1C9FiWr+c" },
+        { "...": "mcU-e_A2tUM4jbm0BNrOTeCgrGUzImXepLonaMnFCvE" }
+    ] 
+}
+
+Disclosures:
+Yr1KUNY... = ["f_ZD3150Og1VUlqREPe_mg","FR"]
+mcU-e_A... = ["YIKG6qOkXXNzMQtsX_H-lw","UK"]
+```
+
+``` ascii-art
+{
+    "family_name", "Möbius",
+    "_sd": [ "wjj8P4DvQ9cNWoDtqkOPYKfV7OCbuxlzkGTdUEGblVw" ]
+}
+
+Disclosures:
+Yr1KUNY... = ["f+ZD3150Og1VUlqREPe+mg", "FR"]
+mcU-e_A... = ["YIKG6qOkXXNzMQtsX_H-lw", "UK"]
+wjj8P4D... = [
+    "4bPeSUK9aY_PlF4gcXnlGg", 
+    "nationalities", 
+    [
+        "DE",
+        { "...": "62bd4a50d61723a5b324a84fe07c32bbd4357a3b9b542709446d42f45896afe7" },
+        { "...": "99c53e7bf036b543388db9b404dace4de0a0ac65332265dea4ba2768c9c50af1" }
+    ]
+]
+```
```

Comment:
what's going on here with the hex encoding (above and below) and the unformatted example outside the triple ticks (below)? Which is sort of a rhetorical question. Please fix.

Reply:
see also https://twitter.com/__b_c/status/434083504381378560

Reply:
It would be great if we could have a generated example here to avoid any inconsistencies.

Reply:
> see also https://twitter.com/__b_c/status/434083504381378560

how many puppies have died since 2014...

Reply:
this ascii-art is different from how disclosures are illustrated in the rest of the spec, should we follow those?:

```
Claim sub:

SHA-256 Hash: X6ZAYOII2vPN40V7xExZwVwz7yRmLNcVwt5DL8RLv4g
Disclosure:
WyIyR0xDNDJzS1F2ZUNmR2ZyeU5STjl3IiwgInN1YiIsICI2YzVjMGE0OS1i
NTg5LTQzMWQtYmFlNy0yMTkxMjJhOWVjMmMiXQ
Contents: ["2GLC42sKQveCfGfryNRN9w", "sub",
"6c5c0a49-b589-431d-bae7-219122a9ec2c"]
```

Reply:
@bifurcation did you want to use sd-jwt library that we use to generate other examples https://github.com/openwallet-foundation-labs/sd-jwt-python
or just copy past from other examples in the spec?

Reply:
tried to save a few cute puppies 21f3ae8e67913798fcb2186ba342efabc5770d01